### DISCUSSION TITLE
Cannot find name 'PopStateEvent'

### DISCUSSION THREAD

**Discussion by smitray** [vercel/next.js#21223] [Help]
**What version of Next.js are you using?**

10.0.4

**What version of Node.js are you using?**

14.15.4

**What browser are you using?**

Chrome

**What operating system are you using?**

Linux

**How are you deploying your application?**

Other platform

**Describe the Bug**

I'm trying to implement Next with Koa server in typescript. But whenever I'm compiling it, it's throwing this error. Error is coming from the following file /next/dist/next-server/lib/router/router.d.ts and line number is 132

**Expected Behavior**

It shouldn't throw any TS2304 error

**To Reproduce**

Complete code you can find it here https://github.com/smitray/next-monorepo and while building it you will receive the error

Hi,
I'm a newbie here and I'm stuck while building the application. I'm getting the error 

```
node_modules/next/dist/next-server/lib/router/router.d.ts:132:21 - error TS2304: Cannot find name 'PopStateEvent'.
```

Kindly help,

Thanks

**Comment by hideokamoto** [comment]
Have you tried to add ` "skipLibCheck": true,` into your `tsconfig.json` ?
I saw a similar error when using AWS CDK (serverless-nextjs), but when I've added the option, the build has succeeded.

By default, the `tsconfig.json` that is made by `create-next-app` is the same setting. So I think we need to set this option.
